### PR TITLE
QtWidgets fixes

### PR DESCRIPTION
**Pull request by petersteneteg**

### CODE REVIEW

File: modules/qtwidgets/src/properties/lightpropertywidgetqt.cpp [hunk 1]
```diff
@@ -73,20 +123,29 @@ LightPropertyWidgetQt::LightPropertyWidgetQt(FloatVec3Property* property)
         LogWarn("Maximum value is assumed to be positive. Widget may produce values out of range.")
     }
 
-    vec3 maxVal = glm::abs(property_->getMaxValue());
-    radiusSpinBox_->setMinimum(-glm::length(maxVal));
-    radiusSpinBox_->setMaximum(glm::length(maxVal));
+    radius_->setMinValue(0, ConstraintBehavior::Immutable);
```
Comment (martinfalk) [inline modules/qtwidgets/src/properties/lightpropertywidgetqt.cpp:126]: immutable?

File: src/core/properties/ordinalproperty.cpp [hunk 2]
```diff
@@ -91,4 +91,14 @@ OrdinalPropertyState<vec3> util::ordinalColor(const vec3& value,
             PropertySemantics::Color};
 }
 
+OrdinalPropertyState<vec3> util::ordinalLight(const vec3& pos, float min, float max,
```
Comment (martinfalk) [inline src/core/properties/ordinalproperty.cpp:94]: isn't that even more generic than light? It's a position...

File: src/qt/editor/welcomewidget.cpp [hunk 3]
```diff
@@ -381,13 +323,7 @@ WelcomeWidget::WelcomeWidget(InviwoMainWindow* window, QWidget* parent)
 
     setTabOrder(filterLineEdit_, filetree_);
     setTabOrder(filetree_, loadWorkspaceBtn_);
-    setTabOrder(loadWorkspaceBtn_, newButton);
```
Comment (martinfalk) [inline src/qt/editor/welcomewidget.cpp:384]: did you check whether the tab order is still as expected?

**Review by martinfalk — DISMISSED** [review]

File: modules/qtwidgets/src/properties/lightpropertywidgetqt.cpp (hunk 1 above)
Comment (petersteneteg) [reply modules/qtwidgets/src/properties/lightpropertywidgetqt.cpp:126]: it's a length can never be less then 0, I've removed the hack with negative distance from the gui at least

File: src/core/properties/ordinalproperty.cpp (hunk 2 above)
Comment (petersteneteg) [reply src/core/properties/ordinalproperty.cpp:94]: it has the light semantics, but I was also considering adding a pos something... but did not have a need for it at the moment...

File: src/qt/editor/welcomewidget.cpp (hunk 3 above)
Comment (petersteneteg) [reply src/qt/editor/welcomewidget.cpp:384]: I just moved them, but I have not checked yet

Comment (petersteneteg) [reply src/qt/editor/welcomewidget.cpp:384]: it still works, but we have no focus indication on the buttons...

File: modules/qtwidgets/src/lightpositionwidgetqt.cpp [hunk 4]
```diff
@@ -36,119 +42,100 @@
...
 
-float LightPositionWidgetQt::getX() const { return sin(theta_) * cos(phi_); }
+double LightPositionWidgetQt::getX() const { return pos_.x; }
 
-float LightPositionWidgetQt::getY() const { return -sin(theta_) * sin(phi_); }
+double LightPositionWidgetQt::getY() const { return -pos_.y; }
 
-float LightPositionWidgetQt::getZ() const { return (radius_ < 0.0f ? -1.0f : 1.0f) * cos(theta_); }
+double LightPositionWidgetQt::getZ() const {
+    return (radius_ < 0.0f ? -1.0f : 1.0f) * std::sqrt(1.0 - glm::length(pos_));
```
Comment (ResearchDaniel) [inline modules/qtwidgets/src/lightpositionwidgetqt.cpp]: If you want to nitpick: remove f in -1.0f and 1.0f

**Review by ResearchDaniel — DISMISSED** [review]
Nice!

**Review by martinfalk — DISMISSED** [review]

File: modules/qtwidgets/src/lightpositionwidgetqt.cpp (hunk 4 above)
Comment (petersteneteg) [reply modules/qtwidgets/src/lightpositionwidgetqt.cpp]: humm missed one... thanks

**Review by martinfalk — APPROVED** [review]

**Pull request merged**